### PR TITLE
fypp 3.1 (new formula)

### PR DESCRIPTION
**Pull request by awvwgk**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Tested in https://github.com/awvwgk/homebrew-fpm/pull/8

### CODE REVIEW

File: Formula/fypp.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  head "https://github.com/aradi/fypp.git", branch: "master"
+
+  depends_on "python@3.9"
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    system "#{bin}/fypp", "--version"
```
Comment (carlocab) [inline Formula/fypp.rb:18]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

File: Formula/fypp.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  desc "Python powered Fortran preprocessor"
+  homepage "https://fypp.readthedocs.io/en/stable/"
+  url "https://github.com/aradi/fypp/archive/refs/tags/3.1.tar.gz"
+  sha256 "0f66e849869632978a8a0623ee510bb860a74004fdabfbfb542656c6c1a7eb5a"
+  license "BSD-2-Clause"
+  head "https://github.com/aradi/fypp.git", branch: "master"
+
+  depends_on "gcc" => :test
+  depends_on "python@3.9"
```
Comment (SMillerDev) [inline Formula/fypp.rb:12]: Does it work with `3.10`?

Comment (awvwgk) [reply Formula/fypp.rb:12]: Sure, should this be bumped to 3.10 or is there something like a version range available?

Comment (SMillerDev) [reply Formula/fypp.rb:12]: This should be bumped to 3.10, all formulae should use the latest dependencies that work.

**Review by SMillerDev — APPROVED** [review]
Thanks @awvwgk  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**